File: shared-pref-sample/src/main/java/ru/surfstudio/android/shared/pref/sample/ui/base/configurator/FragmentScreenConfigurator.java
package ru.surfstudio.android.shared.pref.sample.ui.base.configurator;

import android.os.Bundle;

import ru.surfstudio.android.core.mvp.configurator.BaseFragmentViewConfigurator;
import ru.surfstudio.android.core.ui.activity.CoreActivityInterface;
import ru.surfstudio.android.sample.dagger.ui.base.dagger.screen.DefaultFragmentScreenModule;
import ru.surfstudio.android.shared.pref.sample.ui.base.dagger.activity.ActivityComponent;

/**
 * Базовый конфигуратор для экрана, основанного на фрагменте
 */

public abstract class FragmentScreenConfigurator
        extends BaseFragmentViewConfigurator<ActivityComponent, DefaultFragmentScreenModule> {

    public FragmentScreenConfigurator(Bundle args) {
        super(args);
    }

    @Override
    protected DefaultFragmentScreenModule getFragmentScreenModule() {
        return new DefaultFragmentScreenModule(getPersistentScope());
    }

    @Override
    protected ActivityComponent getParentComponent() {
        return (ActivityComponent) ((CoreActivityInterface) getTargetFragmentView().getActivity())
                .getPersistentScope()
                .getConfigurator()
                .getActivityComponent();
    }
}

File: navigation/sample-standard/src/main/java/ru/surfstudio/android/navigation/sample_standard/di/ui/configurator/FragmentScreenConfigurator.kt
package ru.surfstudio.android.navigation.sample_standard.di.ui.configurator

import android.os.Bundle
import ru.surfstudio.android.core.mvp.configurator.BaseFragmentViewConfigurator
import ru.surfstudio.android.core.ui.activity.CoreActivityInterface
import ru.surfstudio.android.navigation.sample_standard.di.ui.ActivityComponent
import ru.surfstudio.android.sample.dagger.ui.base.dagger.screen.DefaultFragmentScreenModule

/**
 * Базовый конфигуратор для экрана, основанного на фрагменте
 */
abstract class FragmentScreenConfigurator(args: Bundle?) : BaseFragmentViewConfigurator<ActivityComponent?, DefaultFragmentScreenModule?>(args) {
    override fun getFragmentScreenModule(): DefaultFragmentScreenModule {
        return DefaultFragmentScreenModule(persistentScope)
    }

    override fun getParentComponent(): ActivityComponent {
        return (persistentScope.screenState.fragment.activity as CoreActivityInterface)
                .persistentScope
                .configurator
                .activityComponent as ActivityComponent
    }
}